src/test_suite/fixture_utils.py

```diff
@@ -25,7 +24,7 @@ def create_fixture(test_file: Path) -> int:
     serialized_context = read_context(test_file)
     results = process_single_test_case(serialized_context)
 
-    pruned_results = prune_execution_result(serialized_context, results)
+    pruned_results = globals.harness_ctx.prune_effects_fn(serialized_context, results)
```

Comment:
Interface defines deserialized objects, not serialiazed strs. Might want to make one of them consistent with the others